### PR TITLE
Hoplite

### PR DESCRIPTION
**Pull request by dustinsand**
Issue #44 

Loads hocon, java property and yaml configuration files for default, qa and prod.
Displays error message when the configuration is incomplete.

# sksamuel/hoplite
You can find example configuration files in kotlin-jvm/src/main/resources/hoplite/
Load the default Hocon configuration
Test: Config(env=dev, server=Server(port=8080, redirectUrl=/v1/404.html))
Load the qa configuration from a property file cascading over the Hocon defaults
Test: Config(env=qa, server=Server(port=8080, redirectUrl=/v1/404.html))
Load the prod Yaml configuration cascading over the Hocon defaults
Test: Config(env=prod, server=Server(port=443, redirectUrl=/v1/404.html))
Display errors because the qa properties file does not contain the server values
Error loading config because:

    - Could not instantiate 'playground.hoplite.Config' because:

        - 'server': Missing from config

### CODE REVIEW

**Comment by LouisCAD** [comment]
What do you think about showing an example of HOCON format instead of YAML @dustinsand?

**Comment by dustinsand** [comment]
Not a problem.  Added configuration file examples in Hocon, Yaml and Java Property format.

File: versions.properties [hunk 1]
```diff
@@ -29,6 +29,8 @@ version.com.nhaarman.mockitokotlin2..mockito-kotlin=2.2.0
 
 version.com.sksamuel.hoplite..hoplite-core=1.3.7
 
+version.com.sksamuel.hoplite..hoplite-hocon=1.3.7
+
 version.com.sksamuel.hoplite..hoplite-yaml=1.3.7
```
Comment (LouisCAD) [inline versions.properties:34]: This line can now be deleted.

**Comment by LouisCAD** [comment]
Great for the update! I'll let @jmfayard do the rest of the review.

**Review by jmfayard — APPROVED** [review]

**Comment by jmfayard** [comment]
Looks good to me, thanks a lot  @dustinsand !

**Pull request merged**